homeassistant/components/aladdin_connect/cover.py

```diff
@@ -79,27 +80,83 @@ class AladdinDevice(CoverEntity):
     _attr_device_class = CoverDeviceClass.GARAGE
     _attr_supported_features = SUPPORTED_FEATURES
 
-    def __init__(self, acc: AladdinConnectClient, device: DoorDevice) -> None:
+    def __init__(
+        self, acc: AladdinConnectClient, device: DoorDevice, entry: ConfigEntry
+    ) -> None:
         """Initialize the Aladdin Connect cover."""
         self._acc = acc
+
         self._device_id = device["device_id"]
         self._number = device["door_number"]
         self._attr_name = device["name"]
         self._attr_unique_id = f"{self._device_id}-{self._number}"
 
-    def close_cover(self, **kwargs: Any) -> None:
+    async def async_added_to_hass(self) -> None:
+        """Connect Aladdin Connect to the cloud."""
+
+        @callback
+        async def update_callback() -> None:
+            """Schedule a state update."""
+            self.async_write_ha_state()
+
+        self._acc.register_callback(update_callback, self._number)
+        await self._acc.get_doors(self._number)
+        self.async_write_ha_state()
```

Comment:
```suggestion
```

Reply:
I added this code because the system was waiting for the 5 minute polling timer before updating at startup (leaving the cover in "unknown" for 5 minutes).  Is there some other way to fix this?

Reply:
I added this code because at startup it was taking 5 minutes (the poll timer) to go from unknown to "known" state.  Is there a more correct way to make that happen?

Reply:
The entity will always update state at the end of entity addition. So this call isn't needed.

If you want to do a poll (async_update) during entity addition you should call `async_add_entities` with second argument `True`.